testdata/dnn/download_models.py

```diff
@@ -50,63 +60,115 @@ def verify(self):
         except Exception as e:
             print('  verify {}'.format(e))
 
-    def get(self):
+    def cache_copy(self, cache):
+        if not cache:
+            return False
+        try:
+            candidate = cache / self.filename
+            if candidate.is_file():
+                print('  cache {} -> {}'.format(candidate, self.filename))
+                self.prepare_file()
+                shutil.copy(candidate, self.filename)
+                return True
+        except Exception as e:
+            print('  cache {}'.format(e))
+
+    def get_sub(self, cache, arch):
+        print('** {}'.format(self.filename))
+
         if self.verify():
-            print('  hash match - skipping')
             return True
+        if self.cache_copy(cache):
+            return self.verify()
```

Comment:
If there is a file in the cache but with a different SHA (verify returns False) shall we try download file instead?